### PR TITLE
Add schedule notification method

### PR DESCRIPTION
**Pull request by makarski**
Based on https://github.com/Azure/azure-notificationhubs-java-backend/blob/master/NotificationHubs/src/com/windowsazure/messaging/NotificationHub.java#L587, Since no api reference found.

- [x] add `Schedule` tests

### CODE REVIEW

**Comment by makarski** [comment]
@ghophp, @hva, please, review.

**Comment by hva** [comment]
Looks good, :+1:

File: notihub/notihub.go [hunk 1]
```diff
@@ -134,28 +139,44 @@ func NewNotificationHub(connectionString, hubPath string) *NotificationHub {
 		}
 	}
 
-	urlPath := fmt.Sprintf("%s/messages%s", hubPath, apiVersion)
-	aURL := &url.URL{Host: endpoint, Scheme: "https", Path: urlPath}
+	stdPath := fmt.Sprintf("%s/messages%s", hubPath, apiVersion)
+	scheduledPath := fmt.Sprintf("%s/schedulednotifications%s", hubPath, apiVersion)
```
Comment (ghophp) [inline notihub/notihub.go]: there is no `/` missing to separate the `apiVersion`?

File: notihub/notihub.go [hunk 2]
```diff
@@ -166,6 +187,17 @@ func (h *NotificationHub) Send(n *Notification, orTags []string) ([]byte, error)
 		headers["ServiceBusNotification-Tags"] = strings.Join(orTags, " || ")
 	}
 
+	urlStr := h.stdURL.String()
+	if deliverTime != nil && deliverTime.Unix() > time.Now().Unix() {
+		urlStr = h.scheduleURL.String()
+		headers["ServiceBusNotification-ScheduleTime"] = deliverTime.Format("2006-01-02T15:04:05")
```
Comment (ghophp) [inline notihub/notihub.go]: is there a constant on the time package that maybe match this pattern? looks like the RFC3339 no?

File: notihub/notihub.go (hunk 1 above)
Comment (makarski) [inline notihub/notihub.go]: no.

File: notihub/notihub.go (hunk 2 above)
Comment (makarski) [inline notihub/notihub.go]: rfc3339 has a timezone to that.

**Pull request merged**